diff --git a/megatron/core/dist_checkpointing/strategies/base.py b/megatron/core/dist_checkpointing/strategies/base.py
index 35fca1f..e31264c 100644
--- a/megatron/core/dist_checkpointing/strategies/base.py
+++ b/megatron/core/dist_checkpointing/strategies/base.py
@@ -2,6 +2,7 @@

 """ Strategies base interfaces. """

+from __future__ import annotations
 from abc import ABC, abstractmethod
 from collections import defaultdict
 from enum import Enum
