diff --git a/megatron/core/transformer/moe/moe_utils.py b/megatron/core/transformer/moe/moe_utils.py
index dafedfd..0564cf1 100644
--- a/megatron/core/transformer/moe/moe_utils.py
+++ b/megatron/core/transformer/moe/moe_utils.py
@@ -293,7 +293,7 @@ def permute(
         token_indices = (
             torch.arange(num_tokens, device=routing_map.device).unsqueeze(0).expand(num_experts, -1)
         )
-        sorted_indices = token_indices.masked_select(routing_map)
+        sorted_indices = (token_indices.cpu().masked_select(routing_map.cpu())).npu()
 
     # use the mapping to permute the tokens
     permuted_input = tokens.index_select(0, sorted_indices)
