diff --git a/pretrain_gpt.py b/pretrain_gpt.py
index d6c5cd6..469a19d 100644
--- a/pretrain_gpt.py
+++ b/pretrain_gpt.py
@@ -2,6 +2,7 @@
 """Pretrain GPT."""
 
 import torch
+import mindspeed.megatron_adaptor
 from functools import partial
 from contextlib import nullcontext
 import inspect
@@ -253,10 +254,7 @@ def forward_step(data_iterator, model: GPTModel):
 
 
 def is_dataset_built_on_rank():
-    return (
-        mpu.is_pipeline_first_stage() or mpu.is_pipeline_last_stage()
-    ) and mpu.get_tensor_model_parallel_rank() == 0
-
+    return mpu.get_tensor_model_parallel_rank() == 0
 
 def core_gpt_dataset_config_from_args(args):
     tokenizer = get_tokenizer()
